test/extensions/filters/http/header_to_metadata/config_test.cc

```diff
@@ -31,22 +31,26 @@ void testForbiddenConfig(const std::string& yaml) {
                EnvoyException);
 }
 
-// Tests that an empty header is rejected.
-TEST(HeaderToMetadataFilterConfigTest, InvalidEmptyHeader) {
+// Tests that empty (metadata) keys are rejected.
+TEST(HeaderToMetadataFilterConfigTest, InvalidEmptyKey) {
```

Comment:
Also, I think we're missing a test that attempts to configure a rule with both `header` and `cookie` set.

Reply:
One thing that I have been fuzzy about it, where should the test live, config or filter?

Reply:
technically config -- since it should be rejected at config time. but there are some tests that exercise config that are in the filter test file... we can clean it up later tho.

Reply:
Added both the tests below.